fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -248,6 +248,49 @@ class DDTeamCollectionImpl {
 			Optional<Reference<IDataDistributionTeam>> bestOption;
 			std::vector<Reference<TCTeamInfo>> randomTeams;
 
+			if (ddLargeTeamEnabled() && req.keys.present()) {
+				int customReplicas = self->configuration.storageTeamSize;
+				for (auto& it : self->customReplication->intersectingRanges(req.keys.get())) {
+					customReplicas = std::max(customReplicas, it.value());
+				}
+				if (customReplicas > self->configuration.storageTeamSize) {
+					auto newTeam = self->buildLargeTeam(customReplicas);
+					if (newTeam) {
+						if (newTeam->size() < customReplicas) {
+							if (!self->firstLargeTeamFailure.present()) {
+								self->firstLargeTeamFailure = now();
+							}
+							if (now() - self->firstLargeTeamFailure.get() < SERVER_KNOBS->DD_LARGE_TEAM_DELAY) {
+								req.reply.send(std::make_pair(Optional<Reference<IDataDistributionTeam>>(), foundSrc));
+								return Void();
+							}
+							self->underReplication.insert(req.keys.get(), true);
+						} else {
+							self->firstLargeTeamFailure = Optional<double>();
+						}
+						TraceEvent("ReplicatingToLargeTeam", self->distributorId)
```

Comment:
It might be useful to include the requested replicas size in this trace event so that we can quickly compare whether the given team size matches.